commit aead2c0cbffdda9b46d74a998a4c6aeef423b21a
Author: Hauke Mehrtens <hauke@hauke-m.de>
Date:   Wed Jul 13 16:55:20 2016 +0200

    kmodloader: fix lsmod depends output
    
    Without this patch only the first dependency is shown, with this patch
    all module dependencies are show.
    
    Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

diff --git a/kmodloader.c b/kmodloader.c
index ad1f1c0..e32e6af 100644
--- a/kmodloader.c
+++ b/kmodloader.c
@@ -658,15 +658,26 @@ static int main_rmmod(int argc, char **argv)
 static int main_lsmod(int argc, char **argv)
 {
 	struct module *m;
+	char *dep;
 
 	if (scan_loaded_modules())
 		return -1;
 
 	avl_for_each_element(&modules, m, avl)
-		if (m->state == LOADED)
-			printf("%-20s%8d%3d %s\n",
-				m->name, m->size, m->usage,
-				(*m->depends == '-') ? ("") : (m->depends));
+		if (m->state == LOADED) {
+			printf("%-20s%8d%3d ",
+				m->name, m->size, m->usage);
+			if (m->depends && strcmp(m->depends, "-") && strcmp(m->depends, "")) {
+				dep = m->depends;
+				while (*dep) {
+					printf("%s", dep);
+					dep = dep + strlen(dep) + 1;
+					if (*dep)
+						printf(",");
+				}
+			}
+			printf("\n");
+		}
 
 	free_modules();
 
